src/main/java/com/moodysalem/phantomjs/wrapper/PhantomJSSetup.java

```diff
@@ -0,0 +1,215 @@
+package com.moodysalem.phantomjs.wrapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import com.moodysalem.phantomjs.wrapper.beans.OperatingSystem;
+
+public final class PhantomJSSetup{
+
+	protected final static Logger logger = Logger.getLogger(PhantomJSSetup.class.getName());
+
+	// this will store a reference to the executable phantomjs binary after we unzip the resource
+	protected static File PHANTOM_JS_BINARY = null;
```

Comment:
@moodysalem Gotcha. I've made the whole class final, but maybe being able to support re-initialization is something we need. As for the `PHANTOM_JS_BINARY` being final, we would have to initialize it by making a `private function` call correct? Something like so:

`protected static File PHANTOM_JS_BINARY = initialize();`